### DISCUSSION TITLE
Is it possible to install bundle without marking slot active?

### DISCUSSION THREAD

**Discussion by bluecube** [rauc/rauc#927] [Q&A] [answered]
We're using rauc to update two machines that always need to be at the same version. We have an external mechanism that calls `rauc install` on both of them, but in case the update is interrupted (eg loss of power), there is still a window for desynchronization between the two (one of the machines has to copy the bundle over network delaying the install).

What I'd like is for rauc to install to the inactive slot as normal, but don't flip the active slot flag, allowing me to do this operation later with tighter synchronization and less opportunity for mess ups.

Is this possible?

**Comment by a3f** [comment, marked as answer]
There's 35a13bc201ca` ("install: control automatic activation of newly installed slot")`, which could be used for that. AFAIK, post-install handlers are also called before marking freshly written slot active. Note that both still have a race window: You could install both inactive slots, but still have only one system lose power before marking the slot active.

I don't know the specifics of your system, but if it's imperative that the systems are updated in-sync, this needs to be ensured on _every_ startup and systems will rollback if there's a mismatch. Just checking after update won't account for roll-back due to other issues with the update.

**Reply by bluecube** [reply]
Thank  you, the commit you linked seems to be exactly what I was looking for
